lib/beaker/hypervisor/openstack.rb

```diff
@@ -153,6 +153,7 @@ def cleanup_storage vm
       vm.volumes.each do |vol|
         @logger.debug "Deleting volume #{vol.name} for OpenStack host #{vm.name}"
         vm.detach_volume(vol.id)
+        vol.wait_for { ready? }
```

Comment:
I'd hope not or we are a crap cloud provider ;-) The docs say there is a default timeout of 10 minutes, which in fairness is a bit generous